src/reports/models/reportsManager.js

```diff
@@ -65,6 +67,144 @@ module.exports.deleteReport = async (testId, reportId) => {
     await databaseConnector.deleteReport(testId, reportId);
 };
 
+function generateCSVReport(jsonReport){
+    let outputCSV = "";
+    //Header Row
+    let first_entry = true;
+    for (let index in jsonReport["headers"]){
+        if (first_entry){
+            outputCSV+=jsonReport["headers"][index];
+            first_entry = false;
+        }else{
+            outputCSV+=","+jsonReport["headers"][index];
+        }
+    }
+    //Data Rows
+    for (let index in jsonReport["data"]){
+        first_entry = true;
+        outputCSV+='\n';
+        for (let key in jsonReport["data"][index]){
+            if (first_entry){
+                if (jsonReport["data"][index][key] === undefined){
+                    jsonReport["data"][index][key] = 0;
+                }
+                outputCSV+=jsonReport["data"][index][key];
+                first_entry = false;
+            }else{
+                if (jsonReport["data"][index][key] === undefined){
+                    jsonReport["data"][index][key] = 0;
+                }
+                outputCSV+=","+jsonReport["data"][index][key];
+            }
+            
+        }
+    }
+    return outputCSV;
+}
+
+function generateJSONReport(report){
+    let dataExtracted = {"headers":[
+        "timestamp",
+        "timemillis",
+        "median",
+        "p95",
+        "p99",
+        "rps",
+    ],"data":{}}; //Fixed headers
+    
+    let startTime = new Date(report["start_time"]);
+    let data = report["intermediates"];
+
+    //Dynamic status code headers
+    let statusHeaders= new Set();
+
+    for (let index in data){
+        let time_diff = new Date(data[index]["timestamp"])-startTime;
+        let newEntry={
+            "timestamp":data[index]["timestamp"],
+            "timemillis": time_diff - (time_diff % 30000),
+            "median": data[index]["latency"]["median"],
+            "p95": data[index]["latency"]["p95"],
+            "p99": data[index]["latency"]["p99"],
+            "rps": data[index]["rps"]["mean"],
+        }
+        for (let code in data[index]["codes"]){
+            let header = "status_"+code.toString();
+            if (statusHeaders.has(header) === false){
+                statusHeaders.add(header);
+            }
+            newEntry[header] = data[index]["codes"][code];
+        }
+        dataExtracted["data"][newEntry["timemillis"]] = newEntry;
+    }
+    statusHeaders.forEach((entry)=>{
+        dataExtracted["headers"].push(entry);
+    })
+    return dataExtracted;
+}
+
+module.exports.exportReport = async(aggregateReport, fileFormat) => {
```

Comment:
once you add swagger you won't need this validation (the default switch) because filetype should be closed enum and swagger will validate it's valid